### PR TITLE
Mark strings in HTML files to be translated

### PR DESCRIPTION
**Pull request by tienne-B**
Many strings and files have been forgotten about to be translated. This marks most missed strings to be translated.

Went through a lot of the HTML templates to mark strings for translation.
Some questions about the conventions to translate certain things in the Django way:
* Item separators (_i.e._ commas)
* Variables (saw some {{ var }} being enclosed by translation tags without any extra text)
* Strings in embedded `<script` tags (made an error in `scoresheets_list.html`)
* Are date formats hardcoded (_e.g._ for debate scheduling), and if so, is localization automatically dealt with?

More passes may be needed to pluralize some messages, and Vue templates (#744). The `.po` files are not modified here (the French ones will be in another PR with some translations soon enough though :) )

### CODE REVIEW

**Comment by philipbelesky** [comment]
Awesome! I might leave the first two bullets for @czlee to answer. 

I think for strings within inline `<script>` tags it will be best to use the standard django approach so that the translated string is directly rendered into the template/source. If the javascript is in an external stylesheet that becomes a little problematic. There shouldn't be too many (if any?) cases where that happens though. One way would be to assign it as an inline variable within the js block then have the js read from that. 

For the date time formats I'm not sure. I think the standard time formatters should be user-language aware. But I'm not sure if they would, for instance, format a day/month output in a month-first or day-first variant. That said we don't distinguish between different forms of english, and may not actually ever display dates in a day-month format anyway. So this might be something we need to check and see.

**Comment by tienne-B** [comment]
Another few things about `gettext`:
* Gender support? - as `participant_person` has a `gender` field, it could be useful in some circumstances
* Multiple pluralizations? When a string has two or more numbers with words that may require pluralization

**Comment by philipbelesky** [comment]
Sorry, are you talking about `gettext` in a javascript context? Could you clarify also what you mean by "A method to see where strings are in dev on the web interface"?

**Comment by tienne-B** [comment]
No, I'm talking about `gettext` generally.

Should have clarified what I meant by that. It would be to have strings "point" to where they are coded, such as on the final round's ballot entry page, by hovering (or something else) over "In the grand-final rounds of a British Parliamentary format this form...", it would indicate `results/templates/enter_results.html:49` to know where to find the string.

**Comment by czlee** [comment]
I'll catch up on old messages eventually, forgive me:

**Item separators:** As far as I know, no good way to do it. I currently assemble a non-translatable comma-separated list, try to reword it to avoid the final "and", and hope for the best. This doesn't always work. Let me know if you figure it out.

**Variables:** If the _only_ thing inside a translated string is a single variable comprising user-defined text, it shouldn't be translated. So, for example, `{% blocktrans %}{{ name }}{% endblocktrans %}` is nonsense. If you see it, please fix it!

**Date formats:** In general, date and time formats are locality-aware. You can actually see this on the admin tournament overview page (I think), action log dates should change when you switch languages.

There were some complications though, see: https://github.com/TabbycatDebate/tabbycat/issues/513#issuecomment-366570036

Details are in there, but regarding what @philipbelesky said about MM/DD and DD/MM: It's because of the American–Commonwealth distinction/ambiguity on this that I [overrode English date formats](https://github.com/TabbycatDebate/tabbycat/blob/3d4614b7fc6565166bc899892f2d06717750a794/tabbycat/utils/formats/en/formats.py) to spell out the month (e.g. "18 Aug"), always. So in English that's already accounted for.

**Gender support:** As far as I'm aware, this isn't supported by `gettext`. Either we have to do the annoying gender-neutral thing, or we have to implement it ourselves. That aside, `Person.gender` is an optional field and most tournaments don't use it, so we shouldn't rely on it anyway, not even kind of. Its main use is for gender-balancing adjudicator panels, which only major tournaments tend to care about. For our application I don't think it's worth going through what [Facebook had to figure out about a decade ago](https://www.cnet.com/news/facebook-gender-policy-has-grammar-in-mind/) and [build into its translation platform](https://www.facebook.com/help/translations/1055862821149956).

**Multiple pluralizations:** Also isn't supported by `gettext`. I think I had to do this once with "_m_ adjudicators and _n_ speakers", you might have seen it when you were refactoring the private URLs emails. If you can, rephrase to avoid, otherwise, implement custom logic.

**Seeing where strings are on the web interface:** Not easily. Transifex offers the feature, but only as a (prohibitively expensive) paid feature.

Sorry I don't have any better news! Gettext is a good system compared to having nothing, but there are definitely times I've wished it was more sophisticated.
